lighthouse-core/gather/driver.js

```diff
@@ -464,7 +472,16 @@ class Driver {
    */
   async evaluateAsync(expression, options = {}) {
     const contextId = options.useIsolation ? await this._getOrCreateIsolatedContextId() : undefined;
-    return this._evaluateInContext(expression, contextId);
+    return this._evaluateInContext(expression, contextId).catch(async err => {
```

Comment:
can you switch to try/await/catch? `.catch(async err => {})` is confusing :)

Reply:
*can* I? yes, but let me try to convince you why we shouldn't

I personally find it's semantics very subtly deceptive for this situation.

You start off with 

```js
try {
  return this._evaluateInContext(..)
} catch (err) {
  // do stuff
}
```

then you realize that this won't ever catch anything so you change it to

```js
try {
  return await this._evaluateInContext(..)
} catch (err) {
  // do stuff
}
```

but there are linters out there that prevent the `return await <value>` pattern so it's been engrained in folks to think it's unnecessary (even though it is in this case and good linters like eslint won't flag it).

then you might think OK fine I'll assign it to a value first!
```js
try {
  const value = await this._evaluateInContext(..)
  return value
} catch (err) {
  // do stuff
}
```

that sidesteps the `return await` problem but still looks unnecessary to the casual observer

all of this to avoid a `.catch` which, even if it mixes `.then` and `async` styles is very straightforward to understand what's happening :)

Reply:
Haha, I like this response.

I mean, I don't really care what bad linters say, but setting that aside... :)

My issue is that an async function passed as a callback always raises my alert level, making me double check that the callee can handle an async function. I'd definitely get used to it over time for `.then`/`.catch`, but I can't really foresee this pattern taking the world by storm at any point in the future :)

My vote is to not mix and match, so either
- go all in on async/await here, make readers get used to `await` patterns, even returns
- go all in on promises here, converting the `.catch()` callback to not be `async`

Reply:
> I don't really care what bad linters say

To clarify, good linters still flag `return await <some value>` which gets folks in the habit of thinking it's bad. I consider this to be a more serious version of the semicolon problem. 

At least linters can warn when you needed to insert a semicolon, making them unnecessary IMO, but no one can tell if you meant to catch all the errors or just the sync ones!

I will give up here though and go with option 1